onnxruntime/core/providers/cuda/cu_inc/accumulation_type.cuh

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cuda_fp16.h>
+
+namespace onnxruntime {
+namespace cuda {
+
+// specifies the auxiliary type to use for accumulation of the given type
+template <typename T>
+struct AccumulationType;
+template <>
+struct AccumulationType<half> { using type = float; };
+template <>
+struct AccumulationType<float> { using type = float; };
+template <>
+struct AccumulationType<double> { using type = double; };
+
+template <typename T>
+using AccumulationType_t = typename AccumulationType<T>::type;
```

Comment:
struct AccumulateType has been defined in a few files (bias_softmax_impl.cu, softmax.h and reduction_all.cc). Could you help refactor the code to keep only one and remove others?